pkg/dogstatsd/server.go

```diff
@@ -318,22 +326,17 @@ func (s *Server) forwarder(fcon net.Conn, packetsChannel chan listeners.Packets)
 	}
 }
 
-func (s *Server) worker() {
-	// the batcher will be responsible of batching a few samples / events / service
-	// checks and it will automatically forward them to the aggregator, meaning that
-	// the flushing logic to the aggregator is actually in the batcher.
-	batcher := newBatcher(s.aggregator)
-
-	parser := newParser()
-	for {
-		select {
-		case <-s.stopChan:
-			return
-		case <-s.health.C:
-		case packets := <-s.packetsIn:
-			s.parsePackets(batcher, parser, packets)
-		}
-	}
+// Flush flushes all the data to the aggregator to them send it to the Datadog intake.
+// Set waitForSerializer to true to serialize and send the data synchronously.
+func (s *Server) Flush(waitForSerializer bool) {
+	log.Debug("Received a Flush trigger")
+	// make all workers flush their aggregated data (in the batcher) to the aggregator.
+	for _, w := range s.workers {
+		w.flush()
```

Comment:
I don't think this flush implementation provides any guarantee that the `MetricSample` slice that the batcher sends to the aggregator on the channel are actually read from the channel by the aggregator before `aggregator.Flush` is called.

1. Am I missing something?
2. I don't know if this lack of guarantee is actually a problem or not for our use cases. If it's not a problem, I would simply call that out in a comment. If it's potentially a problem, let's discuss how we can address it.